lib/presentation/widget/recipe_flags.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+import '../../core/util/asset_constants.dart';
+import '../../core/util/dimensions.dart';
+
+class RecipeFlags extends StatelessWidget {
+  late final Map<String, bool> recipeProperties;
+
+  RecipeFlags({
+    Key? key,
+    cheap = false,
+    dairyFree = false,
+    glutenFree = false,
+    healthy = false,
+    lowFoodmap = false,
+    popular = false,
+    sustainable = false,
+    vegan = false,
+    vegetarian = false,
+  }) : super(key: key) {
+    recipeProperties = {
+      'cheap': cheap,
+      'dairy_free': dairyFree,
+      'gluten_free': glutenFree,
+      'healthy': healthy,
+      'low_foodmap': lowFoodmap,
+      'popular': popular,
+      'sustainable': sustainable,
+      'vegan': vegan,
+      'vegetarian': vegetarian,
+    };
```

Comment:
You could create this Map inside recipe, so here you need to call a method inside the recipe model like getRecipeProperties